lib/mysql.php

```diff
@@ -55,7 +52,8 @@ function mysql_connect(
 
             $hash = sha1($hostname . $username . $flags);
             /* persistent connections start with p: */
-            if ($hostname{1} !== ':' && isset(\Dshafik\MySQL::$connections[$hash])) {
+            /* don't use a cached link for those */
+            if (!$new && $hostname{1} !== ':' && isset(\Dshafik\MySQL::$connections[$hash])) {
```

Comment:
I would tend to agree except for the fact that SQL Safe Mode also affects the other parameters (they are all to be ignored) and this behavior hasn't been implemented for the other params. (Plus the feature has been completely removed as of PHP 7.2.0.)

If it is desirable to add compatibility for SQL Safe Mode, I would suggest that being its own bug instead of piggybacked here.

Reply:
That's fair!